src/sql-parser/src/ast/defs/value.rs

```diff
@@ -168,6 +168,25 @@ impl TryFrom<Value> for i32 {
     }
 }
 
+impl<X: TryFrom<Value>> TryFrom<Value> for Vec<X> {
+    type Error = anyhow::Error;
+    fn try_from(v: Value) -> Result<Self, Self::Error> {
+        match v {
+            Value::Array(a) => {
+                let mut out = Vec::with_capacity(a.len());
+                for i in a {
+                    out.push(
+                        i.try_into()
+                            .map_err(|_| anyhow::anyhow!("cannot use value in array"))?,
+                    )
+                }
+                Ok(out)
+            }
+            _ => anyhow::bail!("cannot use value as array"),
+        }
+    }
+}
```

Comment:
Oh, it's a bummer that all these `TryFrom` implementations ended up in the parser, because they are all planning concerns! We should introduce a new `TryFromOption` trait whose definition and implementations can all live in the planner instead. That's fine to leave to a follow up PR though.

Reply:
Yeah--will move them into the planner, but wanted to save myself the typing in case this approach got axed.